sim/battle.js

```diff
@@ -77,6 +77,17 @@ class Battle extends Dex.ModdedDex {
 		this.debugMode = format.debug || !!options.debug;
 		this.formatData = {id: format.id};
 
+		for (const rule of this.getRuleTable(format).keys()) {
+			if (rule.startsWith('+') || rule.startsWith('-') || rule.startsWith('!')) continue;
+			let subFormat = this.getFormat(rule);
+			if (subFormat.exists) {
+				let hasEventHandler = Object.keys(subFormat).some(val =>
+					val.startsWith('on') && !['onBegin', 'onValidateTeam', 'onChangeSet', 'onValidateSet'].includes(val)
+				);
+				if (hasEventHandler) this.addPseudoWeather(rule);
+			}
+		}
```

Comment:
This whole block definitely goes at the end of `constructor`, right above the `setPlayer` calls.